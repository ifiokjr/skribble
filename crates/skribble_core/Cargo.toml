[package]
name = "skribble_core"
version = { workspace = true }
authors = { workspace = true }
categories = { workspace = true }
documentation = "https://docs.rs/skribble_core"
edition = { workspace = true }
homepage = { workspace = true }
include = ["src/**/*.rs", "Cargo.toml", "readme.md"]
keywords = ["build", "atomic", "css"]
license = { workspace = true }
readme = "readme.md"
repository = { workspace = true }
rust-version = { workspace = true }
description = "Core features for the skribble atomic css library"

[dependencies]
derivative = { workspace = true }
derive_more = { workspace = true }
doc-comment = { workspace = true }
globset = { workspace = true }
heck = { workspace = true }
indent_write = { workspace = true }
indexmap = { features = ["serde"], workspace = true }
lazy_static = { workspace = true }
lightningcss = { workspace = true }
readonly = { workspace = true }
regex = { workspace = true }
serde = { workspace = true }
serde_json = { workspace = true }
skribble_color = { workspace = true, features = ["serde"] }
thiserror = { workspace = true }
typed-builder = { workspace = true }
vfs = { workspace = true }

[dev-dependencies]
insta = { features = ["json"], workspace = true }
