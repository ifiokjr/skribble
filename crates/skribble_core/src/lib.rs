#![deny(clippy::all)]
#![forbid(clippy::indexing_slicing)]
#![doc = document_features::document_features!()]

doc_comment::doctest!("../readme.md");

pub use config::*;
pub use constants::*;
pub use css::*;
pub use error::*;
pub use macros::*;
pub use plugin::*;
pub use runner::*;
pub use traits::*;
pub use utils::*;
pub use vfs;
pub use vfs::VfsPath;

mod config;
mod constants;
mod css;
mod error;
mod macros;
mod plugin;
mod runner;
mod traits;
mod utils;
