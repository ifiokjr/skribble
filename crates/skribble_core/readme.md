# skribble_core

> core library for skribble

<br />

[![Crate][crate-image]][crate-link] [![Docs][docs-image]][docs-link] [![Status][ci-status-image]][ci-status-link] [![Unlicense][unlicense-image]][unlicense-link]

## Installation

```toml
[dependencies]
skribble_core = "0.0.0"
```

### Usage

```rust
use skribble_core::*;
```

[crate-image]: https://img.shields.io/crates/v/skribble_core.svg
[crate-link]: https://crates.io/crates/skribble_core
[docs-image]: https://docs.rs/skribble_core/badge.svg
[docs-link]: https://docs.rs/skribble_core
[ci-status-image]: https://github.com/ifiokjr/skribble/workflows/ci/badge.svg
[ci-status-link]: https://github.com/ifiokjr/skribble/actions?query=workflow:ci
[unlicense-image]: https://img.shields.io/badge/license-Unlicence-blue.svg
[unlicense-link]: https://opensource.org/license/unlicense
