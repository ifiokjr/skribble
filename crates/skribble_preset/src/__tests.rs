use skribble_core::*;

use super::*;

#[test]
fn default_can_be_added_to_runner() -> AnyEmptyResult {
  let plugin = PresetDefault::builder().build();
  let config: StyleConfig = StyleConfig::builder()
    .plugins(vec![PluginContainer::from(plugin)])
    .build();

  let mut runner = SkribbleRunner::new(config);
  let runner_config = runner.initialize()?;
  insta::assert_debug_snapshot!(runner_config);

  Ok(())
}

#[test]
fn contained() -> AnyEmptyResult {
  let plugin = PresetDefault::builder().build();
  let config: StyleConfig = StyleConfig::builder()
    .plugins(vec![PluginContainer::from(plugin)])
    .build();
  let mut runner = SkribbleRunner::new(config);
  let runner_config = runner.initialize()?;
  let mut classes = Classes::default();
  classes.insert_factories(vec![
    ClassFactory::from_tokens(runner_config, &["contained"]),
    ClassFactory::from_tokens(runner_config, &["lg", "contained"]),
  ]);
  classes.sort_by_class();

  insta::assert_display_snapshot!(classes.to_skribble_css(runner_config)?);

  Ok(())
}
