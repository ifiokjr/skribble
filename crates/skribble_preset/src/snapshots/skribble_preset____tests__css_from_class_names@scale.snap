---
source: crates/skribble_preset/src/__tests.rs
expression: classes.to_skribble_css(runner_config)?
---
/* Generated by Skribble */
@charset "utf-8";
@layer base, alias, default, priority-class;
@property --sk-translate-x {
  syntax: "<length-percentage>";
  inherits: true;
  initial-value: 0;
}
@property --sk-translate-y {
  syntax: "<length-percentage>";
  inherits: true;
  initial-value: 0;
}
@property --sk-rotate {
  syntax: "<angle>";
  inherits: true;
  initial-value: 0;
}
@property --sk-skew-x {
  syntax: "<angle>";
  inherits: true;
  initial-value: 0;
}
@property --sk-skew-y {
  syntax: "<angle>";
  inherits: true;
  initial-value: 0;
}
@property --sk-scale-x {
  syntax: "<number> | <percentage>";
  inherits: true;
  initial-value: 1;
}
@property --sk-scale-y {
  syntax: "<number> | <percentage>";
  inherits: true;
  initial-value: 1;
}
@layer base {
  :root {
    --sk-translate-x: 0;
    --sk-translate-y: 0;
    --sk-rotate: 0;
    --sk-skew-x: 0;
    --sk-skew-y: 0;
    --sk-scale-x: 1;
    --sk-scale-y: 1;
  }
  .scale\:\$50 {
    transform: translate(var(--sk-translate-x), var(--sk-translate-y)) rotate(var(--sk-rotate)) skewX(var(--sk-skew-x)) skewY(var(--sk-skew-y)) scaleX(var(--sk-scale-x)) scaleY(var(--sk-scale-y));
  }
}
@layer default {
  :root {
    --sk-translate-x: 0;
    --sk-translate-y: 0;
    --sk-rotate: 0;
    --sk-skew-x: 0;
    --sk-skew-y: 0;
    --sk-scale-x: 1;
    --sk-scale-y: 1;
  }
  .\$transform-gpu {
    --sk-invalid-css-variable: --sk-invalid-css-variable;
    transform: translate3d(var(--sk-translate-x), var(--sk-translate-y), 0) rotate(var(--sk-rotate) skewX(var(--sk-skew-x)) skewY(var(--sk-skew-y)) scaleX(var(--sk-scale-x)) scaleY(var(--sk-scale-y));
  }
  .scale\:\$50 {
    --sk-scale-x: .5;
    --sk-scale-y: .5;
  }
}

